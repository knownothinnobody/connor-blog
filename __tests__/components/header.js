import React from "react"
import { render, shallow, mount } from "enzyme"

import Header from "../../src/components/header"

describe("Header", () => {
  it("renders correctly", () => {
    expect(render(<Header />)).toMatchSnapshot()
  })

  it("has initial state of is-not-active", () => {
    const wrapper = shallow(<Header />)
    const instance = wrapper.instance()
    expect(wrapper.state("isOpen")).toBe("is-not-active")
  })

  it("has state of is-active after toggle", () => {
    const wrapper = shallow(<Header />)
    const instance = wrapper.instance()
    instance.toggleNavbar()
    expect(wrapper.state("isOpen")).toBe("is-active")
  })

  it("has state of is-not-active after two toggles", () => {
    const wrapper = shallow(<Header />)
    const instance = wrapper.instance()
    instance.toggleNavbar()
    instance.toggleNavbar()
    expect(wrapper.state("isOpen")).toBe("is-not-active")
  })
})
