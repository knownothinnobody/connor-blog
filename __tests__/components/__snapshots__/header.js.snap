// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<nav
  aria-label="main navigation"
  className="navbar is-fixed-top"
  role="navigation"
>
  <div
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item logo-image-holder"
        href="/"
      >
        <img
          alt=""
          aria-hidden="true"
          src="test-file-stub"
        />
        <span
          className="is-sr-only"
        >
          Home
        </span>
      </a>
      <a
        aria-expanded="false"
        aria-label="menu"
        className="navbar-burger burger is-not-active"
        onClick={[Function]}
        role="button"
      >
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
      </a>
    </div>
    <div
      className="navbar-menu is-not-active"
    >
      <div
        className="navbar-start"
      >
        <a
          className="navbar-item"
          href="/"
        >
          Home
        </a>
        <a
          className="navbar-item"
          href="/about"
        >
          About
        </a>
        <a
          className="navbar-item"
          href="contact"
        >
          Contact
        </a>
      </div>
    </div>
  </div>
</nav>
`;
